### ISSUE TITLE
What's the correct way to invoke feature extraction with multiple inputs?

### ISSUE THREAD

**Issue by timwhitlock** [CodeWithKyrian/transformers-php#44]
### Your question

The [docs here](https://codewithkyrian.github.io/transformers-php/feature-extraction#pipeline-input-options) state that you can pass an array of strings for batch processing, but when I do this (like the example with `pooling: mean`) I get different embedding vectors than if I invoke with strings one-by-one. 

Is this because of the pooling value? Should it be set to `"cls"` or `"none"`? Other options don't seem to return array keys corresponding to the input keys.

**Comment by CodeWithKyrian** [comment]
Thank you for bringing this to my attention. I’ve tested the issue and confirmed that there is indeed a discrepancy when processing batch inputs compared to single inputs. The differences are minimal, but they do exist.

From my research, it appears this issue is not unique to TransformersPHP but also affects the original Python Transformers library and Transformers.js. I will keep track of this issue and update the package when a resolution is found.

For now, the small discrepancies should not significantly affect the quality of the embeddings. One trick I noticed is that in batch mode, having the input texts of relatively the same length or similar lengths reduces the discrepancy.

Regarding the pooling value, it does impact the embeddings you receive, but isn't the cause of the issue. If you haven’t done so already, you might want to experiment with different pooling values such as “cls” or “none” to see if they yield more consistent results with your use case. [Here's](https://blog.ml6.eu/the-art-of-pooling-embeddings-c56575114cf8?gi=9edcac20b1c4) an article on the pooling modes and their effects.

Please feel free to reach out if you have any further questions or need more assistance.

**Comment by timwhitlock** [comment]
Thanks a lot for the reply. 

It's true that the difference is small, and batch processing may prove to be worth the performance gain. However, for my current testing purposes I've resorted to doing embeddings one-by-one.

I noticed the discrepancy when comparing a query vector (computed on its own) with an identical input (whose vector had been previously computed in a  batch). They yielded a 99% similarity instead of 100%, which left me scratching my head for a while. A simple string comparison is enough to fix that, and saves a vector computation into the bargain. 

Thanks again for bringing self-hosted language models within reach of PHP developers!
